### PR TITLE
[BEAM-3076] support TIMESTAMP in BeamRecordType

### PR DESCRIPTION
**Pull request by zangshayang1**
This is to support Timestamp datatype in BeamRecordSqlType.java by introducing TimestampCoder. Previously Timestamp datatype is mapped to Date type.

### CODE REVIEW

**Comment by XuMingmin** [comment]
R: @xumingmin

**Comment by XuMingmin** [comment]
retest this please

File: sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamSortRelTest.java [hunk 1]
```diff
@@ -49,16 +50,16 @@ public void prepare() {
...
-            1L, 1, 2.0, new Date(1),
-            2L, 4, 3.0, new Date(2),
-            2L, 1, 4.0, new Date(3),
-            5L, 5, 5.0, new Date(4),
-            6L, 6, 6.0, new Date(5),
-            7L, 7, 7.0, new Date(6),
-            8L, 8888, 8.0, new Date(7),
-            8L, 999, 9.0, new Date(8),
-            10L, 100, 10.0, new Date(9)
+            1L, 2, 1.0, new Timestamp(0),
```
Comment (XuMingmin) [inline sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamSortRelTest.java]: can you check if we've test case to cover `Types.DATE`?

Comment (zangshayang1) [reply sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamSortRelTest.java]: We don't. SoryByTimestamp() is the only test case that calls "order by order_time". Previously, TIMESTAMP is not supported so we used Date type here. If we want to test SortByDate(), we can extend current table with a fifth column named "order_date", just for testing purpose. What do you think? 

PS. The build has some problem. I will update the PR soon.

**Comment by zangshayang1** [comment]
retest this please

**Comment by zangshayang1** [comment]
retest this please

**Comment by zangshayang1** [comment]
retest this please

**Comment by zangshayang1** [comment]
retest this please

**Comment by zangshayang1** [comment]
retest this please

**Comment by zangshayang1** [comment]
retest this please

**Comment by zangshayang1** [comment]
retest this please

**Pull request closed without merging**